app.py

```
@@ -1616,16 +1776,159 @@ def _ensure_nanoleaf_client() -> NanoleafClient:
+            raise ValueError(f'Unhandled Home Assistant batch kind: {unreachable}')
+    except Exception as exc:
+        device = item[1] if len(item) > 1 else None
+        label = getattr(device, 'label', None) or getattr(device, 'entity_id', None) or 'unknown'
+        msg = f"Error in Home Assistant batch send for {label}: {exc}"
+        if dmx_logger:
+            dmx_logger.warning(msg)
+        else:
+            print(msg)
+
+
+def _ha_send_one_batch():
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `_ha_send_one_batch`

Add return type annotation: `None`

(ANN202)